src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViews.component.ts

```diff
@@ -0,0 +1,276 @@
+import { NotebookModel } from 'sql/workbench/services/notebook/browser/models/notebookModel';
+import * as notebookUtils from 'sql/workbench/services/notebook/browser/models/notebookUtils';
+import { IBootstrapParams } from 'sql/workbench/services/bootstrap/common/bootstrapParams';
+import { Action, IActionViewItem } from 'vs/base/common/actions';
+import { LabeledMenuItemActionItem } from 'vs/platform/actions/browser/menuEntryActionViewItem';
+import { Taskbar } from 'sql/base/browser/ui/taskbar/taskbar';
+import { MenuItemAction } from 'vs/platform/actions/common/actions';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+import { INotificationService } from 'vs/platform/notification/common/notification';
+import { onUnexpectedError } from 'vs/base/common/errors';
+import { localize } from 'vs/nls';
+import { Deferred } from 'sql/base/common/promise';
+import { AngularDisposable } from 'sql/base/browser/lifecycle';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { CellType, CellTypes } from 'sql/workbench/services/notebook/common/contracts';
+import { isUndefinedOrNull } from 'vs/base/common/types';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { NotebookViewsExtension } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViewsExtension';
+import { INotebookView } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViews';
+import { NotebookViewsGridComponent } from 'sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsGrid.component';
+
+export const NOTEBOOKVIEWS_SELECTOR: string = 'notebook-view-component';
+
+@Component({
+	selector: NOTEBOOKVIEWS_SELECTOR,
+	templateUrl: decodeURI(require.toUrl('./notebookViews.component.html'))
+})
+
+export class NotebookViewComponent extends AngularDisposable implements INotebookEditor {
+	@Input() model: NotebookModel;
+	@Input() activeView: INotebookView;
+	@Input() views: NotebookViewsExtension;
+
+	@ViewChild('container', { read: ElementRef }) private container: ElementRef;
+	@ViewChild('viewsToolbar', { read: ElementRef }) private viewsToolbar: ElementRef;
+	@ViewChild(NotebookViewsGridComponent) private gridstack: NotebookViewsGridComponent;
+	@ViewChildren(CodeCellComponent) private codeCells: QueryList<CodeCellComponent>;
+	@ViewChildren(TextCellComponent) private textCells: QueryList<TextCellComponent>;
+
+	protected _actionBar: Taskbar;
+	public previewFeaturesEnabled: boolean = false;
+	private _modelReadyDeferred = new Deferred<NotebookModel>();
+
+	private _scrollTop: number;
+
+	constructor(
+		@Inject(IBootstrapParams) private _notebookParams: INotebookParams,
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef,
```

Comment:
You're mixing prefixing with _ and not for private vars, let's be consistent (preference being to using _ for private vars)